stl/inc/atomic

```diff
@@ -73,46 +53,174 @@ extern "C" _NODISCARD char __stdcall __std_atomic_has_cmpxchg16b() noexcept;
+        _Compiler_or_memory_barrier();               \
+        break;                                       \
+    case _Atomic_memory_order_release:               \
+    case _Atomic_memory_order_acq_rel:               \
+    default:                                         \
+        _INVALID_MEMORY_ORDER;                       \
+        break;                                       \
+    }
+
+#define _ATOMIC_STORE_PREFIX(_Width, _Ptr, _Desired)      \
+    case _Atomic_memory_order_relaxed:                    \
+        __iso_volatile_store##_Width((_Ptr), (_Desired)); \
+        return;                                           \
+    case _Atomic_memory_order_release:                    \
+        _Compiler_or_memory_barrier();                    \
+        __iso_volatile_store##_Width((_Ptr), (_Desired)); \
+        return;                                           \
+    default:                                              \
+    case _Atomic_memory_order_consume:                    \
+    case _Atomic_memory_order_acquire:                    \
+    case _Atomic_memory_order_acq_rel:                    \
+        _INVALID_MEMORY_ORDER;                            \
+        /* [[fallthrough]]; */
+
+
+#define _ATOMIC_STORE_SEQ_CST_ARM(_Width, _Ptr, _Desired) \
+    _Memory_barrier();                                    \
+    __iso_volatile_store##_Width((_Ptr), (_Desired));     \
+    _Memory_barrier();
+#define _ATOMIC_STORE_SEQ_CST_INTEL(_Width, _Ptr, _Desired) (void) _InterlockedExchange##_Width((_Ptr), (_Desired));
+#define _ATOMIC_STORE_32_SEQ_CST_INTEL(_Ptr, _Desired) \
+    (void) _InterlockedExchange(reinterpret_cast<volatile long*>(_Ptr), static_cast<long>(_Desired));
+
+#define _ATOMIC_STORE_64_SEQ_CST_IX86(_Ptr, _Desired) \
+    _Compiler_barrier();                              \
+    __iso_volatile_store64((_Ptr), (_Desired));       \
+    _Atomic_thread_fence(_Atomic_memory_order_seq_cst);
+
+#if defined(_M_ARM) || defined(_M_ARM64) || defined(_M_ARM64EC)
+#define _ATOMIC_STORE_SEQ_CST(_Width, _Ptr, _Desired) _ATOMIC_STORE_SEQ_CST_ARM(_Width, (_Ptr), (_Desired))
+#define _ATOMIC_STORE_32_SEQ_CST(_Ptr, _Desired)      _ATOMIC_STORE_SEQ_CST_ARM(32, (_Ptr), (_Desired))
+#define _ATOMIC_STORE_64_SEQ_CST(_Ptr, _Desired)      _ATOMIC_STORE_SEQ_CST_ARM(64, (_Ptr), (_Desired))
+#else // ^^^ ARM32/ARM64/ARM64EC hardware / x86/x64 hardware vvv
+#define _ATOMIC_STORE_SEQ_CST(_Width, _Ptr, _Desired) _ATOMIC_STORE_SEQ_CST_INTEL(_Width, (_Ptr), (_Desired))
+#define _ATOMIC_STORE_32_SEQ_CST(_Ptr, _Desired)      _ATOMIC_STORE_32_SEQ_CST_INTEL((_Ptr), (_Desired))
+#ifdef _M_IX86
+#define _ATOMIC_STORE_64_SEQ_CST(_Ptr, _Desired) _ATOMIC_STORE_64_SEQ_CST_IX86((_Ptr), (_Desired))
+#else // ^^^ x86 / x64 vvv
+#define _ATOMIC_STORE_64_SEQ_CST(_Ptr, _Desired) _ATOMIC_STORE_SEQ_CST_INTEL(64, (_Ptr), (_Desired))
+#endif // x86/x64
+#endif // hardware
+
+extern "C" inline void _Atomic_thread_fence(int _Order) {

```

Comment:
Should `_Atomic_thread_fence()` be `noexcept` to match `atomic_thread_fence()`?

Reply:
yeah, let's just make both this and _Check_memory_order noexcept and skip the macro. When mirroring to vcruntime I'll just manually remove that and extern "C".